### DISCUSSION TITLE
Keycloak and Gokapi

### DISCUSSION THREAD

**Discussion by MrMysterius** [Forceu/Gokapi#82] [Q&A] [answered]
Hello, I'm trying to use **gokapi** with **keycloak** for **oauth/oidc**, but I can't get it to work and want to also know if it is even **possible** with gokapi or not.

Also, normally with stuffed that worked with keycloak for me, you had 3 uri not one:

Gokapi config:
![image](https://user-images.githubusercontent.com/25854814/189459656-8e0cf85a-4301-41bc-9691-1b1fc8e0ddf6.png)

Other Configs:
![image](https://user-images.githubusercontent.com/25854814/189459677-f522fa35-b33a-4bd8-9fde-843af357e78f.png)

So I'm currently leaning more towards not possible...

I tried these endpoints on my keycloak as OAuthProviders:

`https://domain.tld/auth/realms/{realm}/protocol/openid-connect/auth`
`https://domain.tld/auth/realms/{realm}/.well-known/openid-configuration`
`https://domain.tld`

But they all didn't work.

**Comment by Forceu** [comment]
I assume in that case it would be the OIDC_AUTH_URI you need to enter. See the example for Google OIDC configuration below. Otherwise try `https://domain.tld/auth/` or have a look at the [keycloak documentation](https://www.keycloak.org/docs/latest/server_admin/#_oidc_clients).

![image](https://user-images.githubusercontent.com/1593467/189494795-9c39d7b3-ae15-4ad9-9af2-47ea4bea16c9.png)

**Reply by MrMysterius** [reply]
Thanks for the reply, solved it myself in the end.

**Comment by MrMysterius** [comment, marked as answer]
Sorry Forgot to answer but I found it. I set my key cloak root URL wrong (It was `https://domain.tld` now it is `https://domain.tld/auth`), so the issuer would miss match (also the URL in the configuration had no endpoint), and in gokapi it worked with this URL `https://domain.tld/auth/realms/{realm}`.

From this URL it gets the well-known openid configuration (`https://domain.tld/auth/realms/{realm}/.well-known/openid-configuration`). That's what I found out looking up how the go openid implementation (`github.com/coreos/go-oidc/v3/oidc`) does it.
